### PR TITLE
COMMERCE-9390 InfoItemCollectionProvider for Related Products

### PR DESCRIPTION
**Pull request by alessiorendina**
https://issues.liferay.com/browse/COMMERCE-9390

---

Resent from #3538 

@smottal please check my changes, thank you!
@pietrobocale Please re-test it manually and check if you need to change the functional tests

### CODE REVIEW

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:commerce-functional

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

File: modules/apps/commerce/commerce-product-api/src/main/java/com/liferay/commerce/product/util/CPDefinitionHelper.java [hunk 1]
```diff
@@ -53,6 +63,11 @@ public CPDataSourceResult search(
 			int start, int end)
 		throws PortalException;
 
+	public List<CPDefinition> search(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-api/src/main/java/com/liferay/commerce/product/util/CPDefinitionHelper.java:66]: I don't think you need to add this, if in the end you publish the product name

File: modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/data/source/DefinitionLinkTypeCPDataSourceImpl.java [hunk 2]
```diff
@@ -81,37 +75,16 @@ public CPDataSourceResult getResult(
...
-				"definitionLinkType", _cpDefinitionLinkTypeConfiguration.type()
-			).put(
-				"excludedCPDefinitionId", cpCatalogEntry.getCPDefinitionId()
-			).build());
-		searchContext.setCompanyId(_portal.getCompanyId(httpServletRequest));
+		CommerceContext commerceContext =
+			(CommerceContext)httpServletRequest.getAttribute(
+				CommerceWebKeys.COMMERCE_CONTEXT);
+
+		SearchContext searchContext =
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/data/source/DefinitionLinkTypeCPDataSourceImpl.java:82]: why do we need this?

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-product-api/src/main/java/com/liferay/commerce/product/util/CPDefinitionHelper.java [hunk 3]
```diff
@@ -34,6 +39,11 @@
 @ProviderType
 public interface CPDefinitionHelper {
 
+	public SearchContext createCPDefinitionLinkSearchContext(
```
Comment (andrea-ale-sbarra) [inline modules/apps/commerce/commerce-product-api/src/main/java/com/liferay/commerce/product/util/CPDefinitionHelper.java:42]: move this to a new util class CPDefinitionLinkUtil

**Review by andrea-ale-sbarra — CHANGES_REQUESTED** [review]

**Comment by alessiorendina** [comment]
ci:test:sf

**Comment by alessiorendina** [comment]
ci:test:relevant

**Comment by alessiorendina** [comment]
ci:test:sf

**Review by andrea-ale-sbarra — APPROVED** [review]

**Comment by andrea-ale-sbarra** [comment]
ci:test:commerce-functional

**Comment by lorenzolapescara** [comment]
Re-sent at #3565.

**Comment by lorenzolapescara** [comment]
ci:close

**Pull request closed without merging**